### PR TITLE
DA_DP_TC073_Verify that all data profile types are listed in priority order under "Item Type" dropped down menu

### PR DESCRIPTION
**Pull request by taile47711**

### CODE REVIEW

**Comment by taile47711** [comment]
image

File: TADashboard/src/test/java/tests/general_settings/NegativeDPGeneralSettingsTest.java [hunk 1]
```diff
@@ -46,6 +51,26 @@ public void tc069_UserIsUnableToProceedToNextStepOrFinishCreatingDataProfileIfNa
...
+            optionInTypeItem.add("actions");
+            optionInTypeItem.add("interface entities");
+            optionInTypeItem.add("test results");
+            optionInTypeItem.add("test case results");
+            optionInTypeItem.add("test suites");
+            optionInTypeItem.add("bugs");
+            Logger.verify("Items are listed in priority order");
+            dataProfilesGeneralSettingsPage.clickItemTypeComboBox();
+            for (String option : optionInTypeItem) {
+                Assert.assertTrue(dataProfilesGeneralSettingsPage.isOptionVisible(option),
```
Comment (MinhQuan119) [inline TADashboard/src/test/java/tests/general_settings/NegativeDPGeneralSettingsTest.java]: How do you check they are listed in priority order?

If it has `actions` before `data sets`. Is this test failed?

Comment (taile47711) [reply TADashboard/src/test/java/tests/general_settings/NegativeDPGeneralSettingsTest.java:70]: I check based expected result in the test case. And if  `action` before `data set`, actual result will unmatch with expected result so this test failed

File: TADashboard/src/main/java/page_objects/DataProfilesGeneralSettingsPage.java [hunk 2]
```diff
@@ -25,4 +26,13 @@ public void createNewProfile(String name, String type, String relatedData) {
         cbbRelatedData.select(relatedData);
         btnFinish.click();
     }
+
+    public void clickItemTypeComboBox() {
+        cbbItemType.click();
+    }
+
+    public boolean isOptionVisible(String itemType) {
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/DataProfilesGeneralSettingsPage.java]: Select has an available method to get all the options.

**Review by tanmle — CHANGES_REQUESTED** [review]

**Review by MinhQuan119 — APPROVED** [review]

File: TADashboard/src/main/java/page_objects/DataProfilesGeneralSettingsPage.java (hunk 2 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/DataProfilesGeneralSettingsPage.java]: I fixed it. Please check again

**Review by MinhQuan119 — CHANGES_REQUESTED** [review]
Please moving your test to the positive class

**Comment by taile47711** [comment]
image

File: TADashboard/src/test/java/tests/general_settings/PositiveDPGeneralSettingsTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        Logger.step("Go to Data Profiles page.");
+        dashboardPage.selectAdministerMenu("Data Profiles");
+
+        Logger.step("Click Add New link");
+        dataProfilesPage.clickAddNewLink();
+    }
+
+    @Test(description = "Verify that all data profile types are listed in priority order under 'Item Type' dropped down menu")
+    public void tc073_AllDataProfileTypesAreListed() {
+        List<String> optionsInTypeItem = new ArrayList<>();
```
Comment (tanmle) [inline TADashboard/src/test/java/tests/general_settings/PositiveDPGeneralSettingsTest.java]: is this Item Type or Type Item?

**Review by tanmle — CHANGES_REQUESTED** [review]

File: TADashboard/src/test/java/tests/general_settings/PositiveDPGeneralSettingsTest.java (hunk 3 above)
Comment (taile47711) [reply TADashboard/src/test/java/tests/general_settings/PositiveDPGeneralSettingsTest.java]: I fixed it. Please check again

**Comment by taile47711** [comment]
image

File: TADashboard/src/test/java/tests/general_settings/PositiveDPGeneralSettingsTests.java [hunk 4]
```diff
@@ -48,4 +48,24 @@ public void tc072_AllDataProfileTypesAreListedUnderItemTypeDroppedDownMenu() {
...
+        listItemTypes.add("Data Sets");
+        listItemTypes.add("Actions");
+        listItemTypes.add("Interface Entities");
+        listItemTypes.add("Test Results");
+        listItemTypes.add("Test Case Results");
+        listItemTypes.add("Test Suites");
+        listItemTypes.add("Bugs");
+
+        Logger.verify("Items are listed in priority order");
+        dataProfilesGeneralSettingsPage.clickItemTypeComboBox();
```
Comment (tanmle) [inline TADashboard/src/test/java/tests/general_settings/PositiveDPGeneralSettingsTests.java]: check if you need to click on the combobox.

Comment (taile47711) [reply TADashboard/src/test/java/tests/general_settings/PositiveDPGeneralSettingsTests.java]: I fixed it. Please check agian

File: TADashboard/src/main/java/page_objects/DataProfilesGeneralSettingsPage.java [hunk 5]
```diff
@@ -28,6 +29,10 @@ public void createNewProfile(String name, String type, String relatedData) {
         btnFinish.click();
     }
 
+    public void clickItemTypeComboBox() {
```
Comment (MinhQuan119) [inline TADashboard/src/main/java/page_objects/DataProfilesGeneralSettingsPage.java]: this method is un-used

**Review by tanmle — APPROVED** [review]

File: TADashboard/src/main/java/page_objects/DataProfilesGeneralSettingsPage.java (hunk 5 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/DataProfilesGeneralSettingsPage.java]: I fixed it. Please check agian

**Review by MinhQuan119 — APPROVED** [review]

**Pull request merged**